driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -395,20 +395,28 @@ public T get() throws IOException {
 
         @Override
         public void operationComplete(final ChannelFuture future) {
-            if (future.isSuccess()) {
-                channel = channelFuture.channel();
-                channel.closeFuture().addListener(new ChannelFutureListener() {
-                    @Override
-                    public void operationComplete(final ChannelFuture future) {
-                        handleReadResponse(null, new IOException("The connection to the server was closed"));
+            synchronized (NettyStream.this) {
```

Comment:
The root of the bug is a race condition between closing the stream, and completing the asynchronous opening of the channel.  Addressed by synchronized between the two tasks

Reply:
Ack